src/filelock/_api.py

```diff
@@ -76,12 +77,41 @@ class ThreadLocalFileContext(FileLockContext, local):
 class BaseFileLock(ABC, contextlib.ContextDecorator):
     """Abstract base class for a file lock object."""
 
-    def __init__(
+    _instances: ClassVar[WeakValueDictionary[str, BaseFileLock]] = WeakValueDictionary()
+
+    def __new__(  # noqa: PYI034
```

Comment:
On one more Ruff error ignored, if we change the return type to `Self` ruff is happy but mypy fails. I opted to satisfy mypy but ignore the ruff rule instead but perhaps there is a combined solution?